File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countNicePairs = function(nums) {
  const map = {};
  const mod = 1e9 + 7;
  
  for (let num of nums) {
    const diff = num - rev(num);
    map[diff] ? map[diff]++ : map[diff] = 1;
  }
  
  return Object.values(map).reduce((acc, curr) => (acc + Math.floor((curr - 1) * curr / 2)) % mod, 0);
};

const rev = (num) => {
  let ans = 0;
  
  while (num) {
    ans = ans * 10 + (num % 10);
    num = Math.floor(num / 10);
  }
  
  return ans;
}

File: 1814-count-nice-pairs-in-an-array/1814-count-nice-pairs-in-an-array.py
class Solution:
    def countNicePairs(self, nums: List[int]) -> int:
        count = Counter()
        
        for a in nums:
            b = int(str(a)[::-1])
            count[a - b] +=1
        return sum(x*(x-1) // 2 for x in count.values()) % (10**9 + 7)